src/sqlfluff/rules/L045.py

```diff
@@ -49,12 +47,21 @@ def _visit_sources(cls, query: Query):
                 if isinstance(source, Query):
                     cls._visit_sources(source)
 
-    def _eval(self, context: RuleContext) -> Optional[LintResult]:
+    def _eval(self, context: RuleContext) -> EvalResultType:
         if context.segment.is_type("statement"):
             crawler = SelectCrawler(context.segment, context.dialect)
             if crawler.query_tree:
                 # Begin analysis at the final, outer query (key=None).
                 self._visit_sources(crawler.query_tree)
                 if crawler.query_tree.ctes:
-                    return LintResult(anchor=context.segment)
+                    return [
+                        LintResult(
+                            anchor=query.cte_name_segment,
+                            description=f"Query defines CTE "
+                            f'"{query.cte_name_segment.raw}" '
+                            f"but does not use it.",
+                        )
+                        for query in crawler.query_tree.ctes.values()
+                        if query.cte_name_segment
+                    ]
```

Comment:
Now returns a list of `LintResult`, one per unused CTE.

Reply:
I think we should add a test with multiple unused CTEs and confirm multiple errors messages are returned. This will need to be a python test rather than just a YAML test.

It also goes inline with our rules that have a python test when custom error messages are returned to ensure the expected error message is returned.

Reply:
Good point. Coming right up!

Reply:
@tunetheweb: Test added. See new file `test/rules/std_L045_test.py`.